analyzer.go

```diff
@@ -51,7 +51,7 @@ func (a *Analyzer) Analyze() (platform.AnalyzedMetadata, error) {
 
 		if api.MustParse(a.Platform.API()).AtLeast("0.8") {
 			if appMeta.BOM != nil && appMeta.BOM.SHA != "" {
-				if err := a.restorePreviousLayer(appMeta.BOM.SHA); err != nil {
+				if err := a.SBOMRestorer.RestoreFromPrevious(a.PreviousImage, appMeta.BOM.SHA); err != nil {
```

Comment:
```suggestion
				if err := a.SBOMRestorer.RestoreFromImage(a.PreviousImage, appMeta.BOM.SHA); err != nil {
```
This might be better